Aspire Budgeting/Facilities/StateManager.swift

```diff
@@ -2,37 +2,55 @@
 //  StateManager.swift
 //  Aspire Budgeting
 //
-//  Created by TeraMo Labs on 11/9/19.
-//  Copyright © 2019 TeraMo Labs. All rights reserved.
-//
 
 import Foundation
 import os.log
 
-final class StateManager: ObservableObject {
-  enum State: Equatable {
-    case loggedOut
-    case verifiedGoogleUser
-    case authenticatedLocally
-    case localAuthFailed
-    case needsLocalAuthentication
-    case hasDefaultSheet
-  }
+enum AppState: Equatable {
+  case loggedOut
+  case authenticatedExternally
+  case authenticatedLocally
+  case localAuthFailed
+  case needsLocalAuthentication
+  case hasDefaultSheet
+}
+
+protocol AppStateManager {
+  var currentState: AppState { get }
+  var currentStatePublished: Published<AppState> { get }
+  var currentStatePublisher: Published<AppState>.Publisher { get }
+  
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/d909f41c73c54120b2fbb4e96bb79560130cb154/Aspire Budgeting/Facilities/StateManager.swift#L22"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>StateManager.swift:22</code></td>
    </tr>
  </tbody>
</table>